src/service/xhr-impl.js

```diff
@@ -147,6 +149,26 @@ export class Xhr {
     return this.fetch(input, setupInit(opt_init, 'text/plain'));
   }
 
+  /**
+   * A subsitute for the standard response.json(), which may optionally strip a prefix before calling JSON.parse().
+   *
+   * @param {!Response} res fetch response to convert to json.
+   * @param {string|undefined} prefix to strip away.
+   * @return {Promise<*>}
+   */
+  xssiJson(res, prefix) {
+    if (!prefix) {
+      return res.json();
+    }
+
+    return res.text().then(txt => {
+      const stripped = startsWith(txt, dev().assertString(prefix))
+        ? txt.slice(prefix.length)
```

Comment:
Should we `user().warn` if the prefix is missing?

Reply:
Sure

Reply:
@choumx: the presubmit check requires the first parameter to `user().warn()` be either `TAG` or an uppercase string. What do you think I should put here? Should I require the caller of `xssiJson` to pipe through the tag name or to use something like `XSSI-JSON`?

Reply:
How about an optional param that defaults to "XHR"? `fetchAmpCors_` already uses that.

Reply:
I'd rather just do "XHR" then. If a specific usage wants to customize it then can add the arg at that point (YAGNI).